docs/pages/contributing/documentation/how-to-contribute.mdx

```diff
@@ -1,168 +1,367 @@
+1. Switch to the master branch by running the following command:
+    
+    ```code
+    $ git switch master
+    ```
 
-```code
-# If you are an external contributor, create a fork and push to it.
-# This command assumes that you have created an SSH key pair and stored
-# your public key in GitHub.
-$ git remote add fork ssh://git@github.com/my-user/teleport
-$ git rebase origin/master
-$ git push -u fork my-branch
-```
+    You can use a versioned branch if you want to update documentation for a specific previous version of Teleport.
+    However, it most cases, you should use the `master` branch.
 
-Visit https://github.com/gravitational/teleport and create a pull request (PR)
-against your branch. Our continuous integration (CI) pipeline will automatically
-assign reviewers to your PR. The best way to ensure that your PR is merged is to
-respond to reviewer feedback in a timely manner. If feedback goes unaddressed
-your PR may be closed due to inactivity. Please merge your PR as soon as
-possible once you have received all required approvals.
+1. Create a new branch of `gravitational/teleport` for your changes by running a command similar to the following:
 
-## Step 4/4 Backport your changes
+    ```code
+    $ git switch -c my-branch
+    ```
 
-Since our docs site is versioned, we want to ensure that any changes you make to
-the most recent version of our documentation are reflected for the versions of
+    You now have a branch for a specific version of the documentation content.
+    Before making any changes, note how the content is organized so you know where to find the files 
+    to update.
+    
+    Within the `content/*.*` directory in your branch, there's a `docs` subdirectory. 
+    The `content/*.*/docs` subdirectory contains the files and folders used to produce a specific version 
+    of the documentation. 
+    
+    The subdirectory includes the following:
+
+    - The `img/` folder for images used inside the pages.
+    - The `pages/` folder for the Markdown files that contain documentation content. 
+      Every file in this folder is rendered as a page.
+    - The `pages/includes` folder for content that's only rendered when it's included in the body of a page.
+    - The `config.json` file for configuring version information.
+
+    You're now ready to make changes in your own branch using the code editor of your choice.
+
+## Step 3/5. Test changes locally
+
+As you make changes to the content in your local branch, it's helpful to see how the changes will be 
+rendered when the documentation is published.
+
+You can use `yarn` to run a development server that detects changes to the documentation and displays 
+them whenever you manually refresh the page in the browser.
+You can also use `yarn` commands to check for errors and fix common issues.
+
+To test changes in your local environment:
+
+1. Change to the root of the `docs` directory by running the following command:
+    
+    ```code
+    $ cd docs
```

Comment:
If the current working directory is in `content/<version>.x`, you'll need to `cd ../..`. There's also a `docs` directory inside the `content/<version>.x` directory, so this might get a little confusing.